code/modules/cm_aliens/hivebuffs/hivebuff.dm

```diff
@@ -0,0 +1,519 @@
+
+/datum/hivebuff/extra_life/major
+	name = "Major Boon of Plenty"
+	desc = "Increases all xenomorph health by 40 for 10 minutes"
+	tier = HIVEBUFF_TIER_MAJOR
+
+	engage_flavourmessage = "The Queen has imbued us with greater fortitude."
+	duration = 10 MINUTES
+	cost = 2
+	number_of_required_pylons = 2
+	radial_icon = "health_m"
+
+/datum/hivebuff/extra_life/major/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.health_modifier += XENO_HEALTH_MOD_SMALL
+	xeno.recalculate_health()
+
+/datum/hivebuff/extra_life/major/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.health_modifier -= XENO_HEALTH_MOD_SMALL
+	xeno.recalculate_health()
+
+/datum/hivebuff/game_ender_caste
+	name = "His Grace"
+	desc = "A huge behemoth of a Xenomorph which can tear its way through defences and flesh alike. Requires open space to grow."
+	tier = HIVEBUFF_TIER_MAJOR
+
+	is_reusable = TRUE
+	cost = 0
+	special_fail_message = "Only one hatchery may exist at a time."
+	cooldown_duration = 15 MINUTES // This buff ceases instantly so we need to incorporation the spawning time too
+	number_of_required_pylons = 2
+	must_select_pylon = TRUE
+
+/datum/hivebuff/game_ender_caste/New()
+	roundtime_to_enable = GLOB.king_acquisition_time
+
+	return ..()
+
+/datum/hivebuff/game_ender_caste/handle_special_checks()
+	if(locate(/mob/living/carbon/xenomorph/king) in hive.totalXenos)
+		special_fail_message = "Only one King may exist at a time."
+		return FALSE
+
+	var/groundside_humans = 0
+	for(var/mob/living/carbon/human/current_human as anything in GLOB.alive_human_list)
+		if(!(isspecieshuman(current_human) || isspeciessynth(current_human)))
+			continue
+
+		var/turf/turf = get_turf(current_human)
+		if(is_ground_level(turf?.z))
+			groundside_humans += 1
+
+	if(groundside_humans < 12)
+		special_fail_message = "There is too little threat to warrant a King."
+		return FALSE
+
+	return !hive.has_hatchery
+
+/datum/hivebuff/game_ender_caste/on_engage(obj/effect/alien/resin/special/pylon/purchased_pylon)
+	var/turf/spawn_turf
+	for(var/turf/potential_turf in orange(5, purchased_pylon))
+		var/failed = FALSE
+		for(var/x_offset in -1 to 1)
+			for(var/y_offset in -1 to 1)
+				var/turf/turf_to_check = locate(potential_turf.x + x_offset, potential_turf.y + y_offset, potential_turf.z)
+				if(turf_to_check.density)
+					failed = TRUE
+					break
+				if(!turf_to_check.is_weedable())
+					failed = TRUE
+					break
+				var/area/target_area = get_area(turf_to_check)
+				if(target_area.flags_area & AREA_NOTUNNEL)
+					failed = TRUE
+					break
+				for(var/obj/structure/struct in turf_to_check)
+					if(struct.density)
+						failed = TRUE
+					break
+		if(!failed)
+			spawn_turf = potential_turf
+			break
+
+	if(!spawn_turf)
+		engage_failure_message = "Unable to find a viable spawn point for the King."
+		return FALSE
+
+	for(var/obj/effect/alien/resin/special/pylon/pylon as anything in hive.active_endgame_pylons)
+		pylon.protection_level = TURF_PROTECTION_OB
```

Comment:
We could also update_icon the pylon and have it use the active sprite when its in this mode.